zeebe/protocol/src/main/resources/protocol.xml

```diff
@@ -145,14 +149,16 @@
     <field name="protocolVersion" id="4" type="uint16"/>
     <field name="valueType" id="5" type="ValueType"/>
     <field name="intent" id="6" type="uint8"/>
-    <field name="brokerVersion" id="9" type="Version" sinceVersion="2" presence="optional"/>
-    <field name="recordVersion" id="10" type="uint16" sinceVersion="3" presence="optional"/>
+    <field name="brokerVersion" id="7" type="Version" sinceVersion="2" presence="optional"/>
+    <field name="recordVersion" id="8" type="uint16" sinceVersion="3" presence="optional"/>
     <!-- populated when RecordType is COMMAND_REJECTION -->
-    <field name="rejectionType" id="7" type="RejectionType"/>
+    <field name="rejectionType" id="9" type="RejectionType"/>
+    <!-- populated when the client passes an operation reference to the gateway-->
+    <field name="operationReference" id="10" type="uint64" presence="optional"/>
     <!-- populated when RecordType is COMMAND_REJECTION, UTF-8-encoded String -->
-    <data name="rejectionReason" id="8" type="varDataEncoding"/>
+    <data name="rejectionReason" id="11" type="varDataEncoding"/>
     <!-- populated when RecordType is COMMAND -->
-    <data name="authorization" id="11" type="varDataEncoding" sinceVersion="4"/>
+    <data name="authorization" id="12" type="varDataEncoding" sinceVersion="4"/>
   </sbe:message>
 
   <sbe:message name="BrokerInfo" id="201" description="Broker topology information">
```

Comment:
AFAIK this would break backwards compatibility. 

SBE is sensitive to order, type and IDs afaik. We had an issue before with doing such that resulted in us not being able to read the rejection messages anymore from old records, see https://github.com/camunda/zeebe/issues/14957

You're only allowed to add new fields at the end afaik (and marking as optional ? 🤔 ) I guess would be also important to increment the version to detect such change. Better to read the SBE guides for such 

> SBE supports the extension of messages via versioned templates. To be backwards compatible a template must only be extended at the end of a block. Messages cannot remain backwards compatible if existing fields are modified or removed. SBE is a fixed format encoding that supports new optional fields being added.

https://github.com/real-logic/simple-binary-encoding/wiki/Message-Versioning

Reply:
Thank you!

I reverted the altered ids, updated the protocol version and added `sinceVersion` to the new field.

Are there any other backward compatible issues that I should be aware of before proceeding with such change?